### ISSUE TITLE
Question about running environments.

### ISSUE THREAD

**Issue by MenggeLiu** [danliu2/caat#15]
Some issues mentioned that using the old version of torch+fairseq can run the training smoothly, and there will be no nan in the training. But after trying I found that the caat model depends on the newer fairseq (>0.12.0). Can you provide a more detailed environment description, including the specific versions of cuda python pytorch fariseq and other dependencies? @danliu2

**Comment by danliu2** [comment]
These experiments are so long ago, I can't remember the details. In those days my torch version is roughly 1.0, and fairseq 0.9, with CUDA8. These may be not critical for model training. However, I think these should not have a big impact on model training. What was mentioned before is just function calling errors caused by the difference of fairseq inference interface.

**Comment by MenggeLiu** [comment]
It is basically confirmed that the loss=nan in the training is a problem of wraprnnt compilation. When I use cuda11 on the a100 machine, I will encounter nan loss in training. When I try to use cuda10 on v100 machine, the loss value is quite normal.

As a100 gpu only support cuda11, would like to ask if there are some solutions to successfully compile warprnnt in cuda11 environment? @danliu2

**Comment by danliu2** [comment]
I suspect there are some outdated practices in cuda kernel code that can be found and resolved by comparing the differences with the latest warprnnt and debugging kernel functions.

I am very sorry that I have not maintained these codes since my graduation when I was doing other work. If you fix these bugs in your work, you are more than welcome to provide PR or redistribute them directly in your repo.
